nostalgia/sources/spotify.py

```diff
@@ -0,0 +1,29 @@
+import just
+import pandas as pd
+
+from nostalgia.ndf import NDF
+from nostalgia.times import datetime_from_format
+
+
+class Spotify(NDF):
+    @classmethod
+    def load(cls, file_path="/Users/nikolay_vasilishin/nostalgia_data/input/spotify/StreamingHistory0.json", nrows=None):
+        spotify = pd.DataFrame(
+            [
+                (
+                    datetime_from_format(x["endTime"], "%Y-%m-%d %H:%M"),
+                    x["trackName"],
+                    x["artistName"],
+                    x["msPlayed"] / 1000
```

Comment:
It's cool. Like this it is duration, but what we would also want is:

```python
spotify["end"] = spotify["time"] + pd.to_timedelta(x["msPlayed"], unit="ms")
```

Because then it becomes a Period (2 times) rather than a Moment (1 time)!

Reply:
Alright, thanks
Maybe we should consider having some helper function for it or annotations

Reply:
I think it's tough to make it smaller than this?